packages/firebase_snippets_app/lib/snippets/auth.dart

```diff
@@ -0,0 +1,622 @@
+      for (final providerProfile in user.providerData) {
+        // ID of the provider (google.com, apple.cpm, etc.)
+        final provider = providerProfile.providerId;
+
+        // UID specific to the provider
+        final uid = providerProfile.uid;
+
+        // Name, email address, and profile photo URL
+        final name = providerProfile.displayName;
+        final emailAddress = providerProfile.email;
+        final profilePhoto = providerProfile.photoURL;
+      }
+    }
+    // [END manage_users_get_provider_specific_profile]
+
+    // [START manage_users_update_profile]
+    await user?.updateDisplayName("Jane Q. User");
+    await user?.updatePhotoURL("https://example.com/jane-q-user/profile.jpg");
+    // [END manage_users_update_profile]
+
+    // [START manage_users_set_email]
+    await user?.updateEmail("janeq@example.com");
+    // [END manage_users_set_email]
+
+    // [START manage_users_send_verification_email]
+    await user?.sendEmailVerification();
+    // [END manage_users_send_verification_email]
+
+    // [START manage_users_set_localization_then_send_email]
+    await FirebaseAuth.instance.setLanguageCode("fr");
+    await user?.sendEmailVerification();
+    // [END manage_users_set_localization_then_send_email]
+
+    const String newPassword = 'goodpassword';
+
+    // [START manage_users_set_new_password]
+    await user?.updatePassword(newPassword);
+    // [END manage_users_set_new_password]
+
+    // [START manage_users_send_reset_email]
+    await FirebaseAuth.instance
+        .sendPasswordResetEmail(email: 'user@example.com');
+    // [END manage_users_send_reset_email]
+
+    // [START manage_users_set_language_code]
+    await FirebaseAuth.instance.setLanguageCode("fr");
+    // [END manage_users_set_language_code]
+
+    final userCredential = await FirebaseAuth.instance.signInAnonymously();
+    final credential = userCredential.credential!;
+
+    // [START manage_users_re_auth]
+    // Prompt the user to re-provide their sign-in credentials.
+    // Then, use the credentials to reauthenticate:
+    await user?.reauthenticateWithCredential(credential);
+    // [END manage_users_re_auth]
+
+    // [START manage_users_delete_user]
+    await user?.delete();
+    // [END manage_users_delete_user]
+  }
+
+  void passwordAuth_createPasswordBasedAccount() async {
+    const String emailAddress = 'user@example.com';
+    const String password = 'sdf8798324jf673qf8h!';
+
+    // [START password_auth_create_password_based_account]
+    try {
+      final credential =
+          await FirebaseAuth.instance.createUserWithEmailAndPassword(
+        email: emailAddress,
+        password: password,
+      );
+    } on FirebaseAuthException catch (e) {
+      if (e.code == 'weak-password') {
+        print('The password provided is too weak.');
+      } else if (e.code == 'email-already-in-use') {
+        print('The account already exists for that email.');
+      }
+    } catch (e) {
+      print(e);
+    }
```

Comment:
Seeing this block of error handling makes me really happy, as this is a pretty common question on Stack Overflow (and we even did an episode of the Boring Show about it). 👏